include/net/nrf_cloud_rest.h

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file nrf_cloud_rest.h
+ *
+ * @brief nRF Cloud REST API.
+ * @defgroup nrf_cloud_rest nRF Cloud REST API
+ * @{
+ *
+ */
+#ifndef NRF_CLOUD_REST_H__
+#define NRF_CLOUD_REST_H__
+
+#include <zephyr/types.h>
+#include <net/nrf_cloud.h>
+#include <net/nrf_cloud_pgps.h>
+#include <net/nrf_cloud_cell_pos.h>
+#include <modem/lte_lc.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief HTTP status codes */
+enum http_status {
+	HTTP_STATUS_UNHANDLED = -1,
+	HTTP_STATUS_NONE = 0,
+	HTTP_STATUS_OK = 200,
+	HTTP_STATUS_ACCEPTED = 202,
+	HTTP_STATUS_PARTIAL = 206,
+	HTTP_STATUS_BAD_REQ = 400,
+	HTTP_STATUS_UNAUTH = 401,
+	HTTP_STATUS_FORBIDDEN = 403,
+	HTTP_STATUS_NOT_FOUND = 404,
+	HTTP_STATUS_BAD_RANGE = 416,
+	HTTP_STATUS_UNPROC_ENTITY = 422,
+};
+
+/** @brief nRF Cloud AGPS REST request types */
+enum nrf_cloud_rest_agps_req_type {
+	NRF_CLOUD_REST_AGPS_REQ_ASSISTANCE,
+	NRF_CLOUD_REST_AGPS_REQ_LOCATION,
+	NRF_CLOUD_REST_AGPS_REQ_CUSTOM,
+	NRF_CLOUD_REST_AGPS_REQ_UNSPECIFIED,
+};
+
+/** @brief Parameters and data for using the nRF Cloud REST API */
+struct nrf_cloud_rest_context {
+
+	/** Connection socket */
+	int connect_socket;
+	/** If the connection should remain after API call */
+	bool keep_alive;
+	/** Timeout value for receiving response data */
+	int32_t timeout_ms;
+	/** Authentication string: JWT or API token */
+	char *auth;
+	/** User allocated buffer for receiving API response.
+	 * Buffer size should be limited according to the
+	 * maximum TLS receive size of the modem.
+	 */
+	char *rx_buf;
+	/** Size of rx_buf */
+	size_t rx_buf_len;
+	/** Fragment size for downloads, set to zero to use
+	 * CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE.
+	 * The rx_buf must be able to hold the https headers
+	 * plus this fragment size.
```

Comment:
total payload (rx_buff) = fragment_size + https headers
i used fragment size since that is the name used in download_client.

i'm fine with changing it to make it more clear.  changing it to payload doesn't make sense to me.
other suggestions?

Reply:
Yeah, you're right, the "fragment" term is sued elsewhere in NCS, so probably best to keep it in line with that. I'm just confusing myself with "TLS fragment size" which is HTTP headers + HTTP fragment